### DISCUSSION TITLE
esp32cam - Timeout waiting for VSYNC error 0x20003

### DISCUSSION THREAD

**Discussion by mit-vincode** [espressif/arduino-esp32#6460] [General]
Hello! I'm doing the ESP32CAM firmware and trying to get the IP address via SerialPort further. I do the firmware through the programmer, grounded IO0, pressed RST at the time of filling. The sketch is filled in correctly on the board. Next, I open the SerialPort Arduino IDE, press RST and expect to get an IP.
1) If IO0 is connected to the ground, it hangs on the message:
rst:0x1 (POWERON_RESET),boot:0x3 (DOWNLOAD_BOOT(UART0/UART1/SDIO_REI_REO_V2))
waiting for download

2) If IO0 is disconnected from the ground, it first passes normally to entry 0x400806b4, then the error crashes: Camera init failed with error 0x20003
What am I doing wrong?

Error when IO0 is disconnected from the ground:
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1216
ho 0 tail 12 room 4
load:0x40078000,len:10944
load:0x40080400,len:6360
entry 0x400806b4

[E][camera.c:222] skip_frame(): ‘
[E][camera.c:1406] esp_camera_init(): Camera init failed with error 0x20003
Camera init failed with error 0x20003

This question has already been asked here before, but I have not found an answer. Thank you in advance.

**Comment by Jason2866** [comment]
GPIO0 zero must be disconnected from any wire after programming because this pin drives the cam clock and does not tolerate any capacitive load

**Reply by easytarget** [reply]
As @Jason2866 says, if you have an AI-THINKER board (or ArduCam) you must let GPIO 0 float after programming. 
Most other boards avoid this mistake and assign a different pin, if in doubt look at the pin definitions in the code to see which number the camera `xclk`  is assigned to.

**Reply by mit-vincode** [reply]
I did it, the IO0 pin is disconnected. However, when you click the Reset button in the Serial Port, an error occurs

[E][camera.c:222] skip_frame(): ‘
[E][camera.c:1406] esp_camera_init(): Camera init failed with error 0x20003
Camera init failed with error 0x20003

**Comment by Jason2866** [comment]
Try [this](https://github.com/arendst/Tasmota/releases) firmware it is verified working well and uses latest Arduino Core from here.
You can easily flash via Webbrowser (Chrome or Edge) from here [https://tasmota.github.io/install/](https://tasmota.github.io/install/)

**Reply by mit-vincode** [reply]
Guys - it worked for me! Thank you guys! Thank you very much))

**Reply by mit-vincode** [reply]
However, in continuation of the topic. I changed the camera to another one - after pressing RST an error occurred:
E (993) camera: JPEG format is not supported on this sensor
Camera init failed with error 0x106
How to fix it?

**Reply by Jason2866** [reply]
You have to use a different webcam sketch. Tasmota webcam does not support cameras without JPEG support.
